### ISSUE TITLE
Provide a way for asserting current path with timeout

### ISSUE THREAD

**Issue by donv** [jnicklas/capybara#1460]
Hi!

We currently use Capybara for testing navigation using links.  A typical use is

```ruby
click_link('Next')
assert_equal '/next_page', current_path
```

This works well most of the time, but it seems the assert + current_path does not wait until the page is loaded.

Searching for a solution, I have found suggestions to wait for some ***content*** on the next page, but that is not what we want.  The content can change over time while in our case the paths are stable.

Is there a way to assert the current_path using the regular timeout?

If not, can it be added?

Here is our current implementation, free for all to use :smile: :

```ruby
  def assert_current_path(path)
    start = Time.now
    sleep 0.1 while (Time.now - start) < Capybara.default_wait_time && current_path != path
    assert_equal path, current_path
  end
```

Is there a better way?

**Comment by twalpole** [comment]
IIRC this has been mentioned before, and it was agreed we should have some sort of path matcher, however it was never decided whether it needs to match Issues with it are whether or not it needs to match query paramters, fragments, etc

**Comment by twalpole** [comment]
Please look at PR #1567 and see if you think that would fit your needs

**Comment by twalpole** [comment]
Fixed through PR #1567

**Comment by donv** [comment]
Sorry for the late response.  This looks good to me :+1:
